app/Http/Controllers/Api/AssetsController.php

```diff
@@ -379,14 +379,25 @@ public function showByTag(Request $request, $tag)
             $assets = $assets->withTrashed();
         }
 
-        $assets = $assets->get();
+        if (($assets = $assets->get()) && ($assets->count()) > 0) {
+
+            // If there is only one result, we should pull the first (and only) asset from the returned collection, since
+            // transformAsset() expects an Asset object, NOT a collection
+            if ($assets->count() == 1) {
```

Comment:
I still think that if you have deleted = true, this branch is going to get taken if there is exactly one deleted asset.